Mta 5505 (#4344)
scaleway/docs-content: pages/key-manager/api-cli/create-dek-api-cli.mdx

        ```

    <Message type="important">
     For convenience, Key Manager returns the plaintext version of the DEK, but it should never be stored in this form. Storing the decrypted plaintext of your DEK undermines the security provided by Key Manager, rendering it ineffective.
    </Message>

## Generate a DEK using the API

Paste the following command to create your data encryption key via the Key Manager API. Make sure that you replace the placeholder values with your own.
        ```bash
        curl --location 'https://api.scaleway.com/key-manager/v1alpha1/regions/fr-par/keys/<your_key_id>/generate-data-key' \
        --header 'Content-Type: application/json' \
        --header 'X-Auth-Token: <your_secret_key>' \
        --data '{
            "algorithm": "aes_256_gcm"
        }'
        ```

Key Manager also supports the `GenerateDataKey` request without a plaintext operation, which only returns an encrypted data encryption key.

If you need to use your DEK, you can decrypt it using the [Decrypt data operation](https://www.scaleway.com/en/developers/api/key-manager/#path-keys-decrypt-data) specifying the `kek_id` parameter used to encrypt it.

> **Key Manager does not handle direct data encryption**. It is specifically designed to **encrypt and decrypt Data Encryption Keys (DEKs) and is limited to processing inputs up to 64KB in size**.

Reviewer: ```suggestion
**Key Manager does not handle direct data encryption**. It is specifically designed to **encrypt and decrypt Data Encryption Keys (DEKs) and is limited to processing inputs up to 64 KB in size**.
```

However, you can use the DEK independently from Key Manager, for example with the [Tink extension](replace-with-link) or with [OpenSSL](replace-with-link).

    <Message type="important">
    The way the KEK is generated, its length, and the encryption algorithm used, **cannot be changed or customized after creation**. However, unlike the KEK, you have the flexibility to choose any encryption algorithm (cipher) you prefer for encrypting and decrypting your data with the DEK. You are not restricted to a specific encryption method for the data itself.

>     **We highly recommend that you use standard and well-established ciphers (and the proper mode), as well as a library like Tink, that chooses the right cryptosystem according to your use-case.**

Reviewer: ```suggestion
    **We highly recommend that you use standard and well-established ciphers (and the proper mode), as well as a library like Tink, that chooses the right cryptosystem according to your use case.**
```

    </Message>

> 
> 

Reviewer: ```suggestion
```
